ESPLoader.js

```diff
@@ -1586,6 +1587,29 @@ class ESPLoader {
         this.log("Device: "+((flashid >> 8) & 0xff).toString(16) + flid_lowbyte.toString(16));
         this.log("Detected flash size: " + this.DETECTED_FLASH_SIZES[flid_lowbyte]);
     }
+
+    hard_reset = async() => {
+        this.transport.setRTS(true);  // EN->LOW
+        await this._sleep(100);
+        this.transport.setRTS(false);
+    }
+
+    soft_reset = async() => {
+        if (!this.IS_STUB) {
+            // 'run user code' is as close to a soft reset as we can do
```

Comment:
If we offer an API it should do the thing, not be "as close to". So in that case this method should raise unless it's ESP8266 + Stub.

Reply:
The comment comes directly from `esptool.py`. The reason is that in the ROM bootloader there is no way to perform a reset and so just running the user code is as close to a reset as you can get given the limitations.
Perhaps the comment should be
```
// The ROM bootloader has to reset function so we just run the user code
```

Reply:
Didn't realize this was ported from ESPTool 👍